src/commands/application/instances/connect.rs

```diff
@@ -43,83 +64,73 @@ pub async fn handle_connect(context: Context, name: &str, port: &u16) -> Result<
 
     let application = context.state.application.unwrap();
 
+    // Check for permission:
     if !has_permission(&client, &application, &namespace, &version)
         .await
         .unwrap()
```

Comment:
```suggestion
    if !has_permission(&client, &application, &namespace, &version)
        .await?
```

I totally missed this, this should be handled.